dvc/repo/metrics/show.py

```diff
@@ -66,13 +66,42 @@ def _read_typed_metric(typ, xpath, fd):
     return ret
 
 
+def _format_csv(content, delimiter):
+    import csv
+
+    reader = csv.reader(content.split("\n"), delimiter=builtin_str(delimiter))
```

Comment:
I'm not sure I understand this. Why do we need this `split('\n')` here?

Reply:
@shcheklein , check this out:

```python
import csv

print("Without splitting new line")
print("--------------------------")
for line in csv.reader("x,y\n1,2"):
    print(line)

print("")
print("Splitting")
print("---------")
for line in csv.reader("x,y\n1,2".split("\n")):
    print(line)

print("")
print("File with new lines on entries")
print("------------------------------")
with open("file.csv", "w") as fobj:
    fobj.write('x,y\n"multi\nline\nstring","single-line"')

with open("file.csv", "r") as fobj:
    for line in csv.reader(fobj):
        print(line)

print("")
print("Content with new lines on entries")
print("---------------------------------")
for line in csv.reader('x,y\n"multi\nline\nstring","single line"'.split("\n")):
    print(line)
```

```
Without splitting new line
--------------------------
['x']
['', '']
['y']
[]
['1']
['', '']
['2']

Splitting
---------
['x', 'y']
['1', '2']

File with new lines on entries
------------------------------
['x', 'y']
['multi\nline\nstring', 'single-line']

Content with new lines on entries
---------------------------------
['x', 'y']
['multilinestring', 'single line']
```

This is because the CSV reader expects an iterator as the first argument.
You don't need to "split" anything when you use a file object (e.g. `with open('file.csv', 'r') as fobj: csv.reader(fobj)`), because python's file object can be used as an iterator (under the hood, the iterator is calling something like `readline()`, and it can deal with "quoted" new lines).

Reply:
Makes sense! I didn't know about `csv.reader` expecting an iterator. Consider using a safer choice to split lines - https://docs.python.org/2/library/stdtypes.html#str.splitlines

Reply:
How about this, btw:

```>>> for line in csv.reader(StringIO('"multi\nline\nstring","single line"\nx,y')):
...     print(line)
...
['multi\nline\nstring', 'single line']
['x', 'y']```?

Reply:
Using StrinIO seems a better way of doing this. Please, test all this stuff on unicode as well. Just to make sure that we don't need to provide utf-8 encoding on Python2 or something like that.

Reply:
:+1:  Awesome, @shcheklein , will fix it on the next patch